packages/block-library/src/playlist/edit.js

```diff
@@ -0,0 +1,212 @@
+/**
+ * External Dependencies
+ */
+import { uniqBy } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import {
+	FormFileUpload,
+	IconButton,
+	Placeholder,
+	ServerSideRender,
+	Toolbar,
+	withNotices,
+	CheckboxControl,
+	PanelBody,
+	SelectControl,
+} from '@wordpress/components';
+import { Component, Fragment } from '@wordpress/element';
+import {
+	BlockControls,
+	InspectorControls,
+	mediaUpload,
+	MediaUpload,
+} from '@wordpress/editor';
+
+class PlaylistEdit extends Component {
+	constructor() {
+		super( ...arguments );
+
+		this.initializePlaylist = this.initializePlaylist.bind( this );
+		this.uploadFromFiles = this.uploadFromFiles.bind( this );
+		this.onUploadFiles = this.onUploadFiles.bind( this );
+		this.onSelectMedia = this.onSelectMedia.bind( this );
+		this.getMimeBaseType = this.getMimeBaseType.bind( this );
+	}
+
+	/**
+	 * The function is used after the editor has enqueued playlist dependencies.
+	 *
+	 * @param  {function} initializePlaylist Function that Initializes playlist
+	*/
+
+	initializePlaylist() {
+		window.wp.playlist.initialize();
```

Comment:
@youknowriad, this block depends on both WordPress global `wp.playlist` and `ServerSideRender` component which is very backend tied. We need to do something about it. There should be reusable blocks which work everywhere and blocks which are reusable but WordPress specific. If we would draw this distinction we wouldn't have to worry about `Code` and `Freeform` anymore.

Reply:
yes, a first step would be to move it ouside of the packages like `html` or `freeform`. Later on, we'd need to rewrite `wp.playlist` as a Gutenberg package and then move the block to the packages as well.